Feature/searchbar

Niv-Regev-Moveo opened the pull request.

Commit:
PageHeader and SearchBar improvements .
now start working on the search bar logic

Niv-Regev-Moveo committed:
search bar now can give you the information about the pokemon that you chose.

still need to keep this information and pass it in to a new linked page that will show the more "detailed" card.

Niv-Regev-Moveo committed:
pr

yuvalkom-M commented on pokedex/src/components/DetailedCard/DetailedCard.tsx:
```diff
@@ -0,0 +1,48 @@
+            <p>Defense: {pokemon.stats[2].base_stat}</p>
+          </div>
+          <div>
+            <br></br>
```
Less common to use </br>. You can give margin-top if needed

yuvalkom-M commented on pokedex/src/components/FavoritePage/FavoritePage.tsx:
```diff
@@ -0,0 +1,5 @@
+const FavoritePage = () => {
+  return <h1>hello favorite page</h1>;
```
remove unnecessary files (all files related to favorite page)

yuvalkom-M commented on pokedex/src/components/HomePage/HomePage.tsx:
```diff
@@ -0,0 +1,52 @@
+function HomePage() {
+  const [pokemonsData, setPokemonsData] = useState<any[]>([]);
+  const [loading, setLoading] = useState(true);
+  const [url, setUrl] = useState(BASE_URL);
```
why use useState for url? (no use of setUrl)

yuvalkom-M commented on pokedex/src/components/PokeSearchResults/PokeSearchResults.tsx:
```diff
@@ -1,12 +1,51 @@
+
+  const onSelect = (selectedName: string) => {
+    const selected = pokemonsData.find(
+      (pokemon: any) => pokemon.name === selectedName
```
don't use any as type

yuvalkom-M commented on pokedex/src/components/PokeSearchResults/PokeSearchResults.tsx:
```diff
@@ -1,12 +1,51 @@
     <div>
-      <h2>Search Results</h2>
-      <ul>
-        {pokemonsData.map((result: any, id: any) => (
```
don't use any as type

yuvalkom-M commented on pokedex/src/components/PokeSearchResults/PokeSearchResults.tsx:
```diff
@@ -1,12 +1,51 @@
+          <option value="" disabled>
+            Select a Pokémon
+          </option>
+          {pokemonsData.map((result: any, id: any) => (
```
same don't use any as type

yuvalkom-M commented on pokedex/src/components/PokemonList/Pokemonlist.tsx:
```diff
@@ -1,6 +1,14 @@
 
-const PokemonList = ({ pokemonsData, loading }: any) => {
+type typePokemonListProps = {
+  pokemonsData: any[];
```
don't use any as type

yuvalkom-M commented on pokedex/src/utils/api.ts:
```diff
@@ -0,0 +1,37 @@
+  }
+}
+
+export async function getSpecificPokemon(urls: string[]): Promise<any[]> {
```
don't use any as type

yuvalkom-M commented on pokedex/src/components/NavBar/NavBar.tsx:
```diff
@@ -10,19 +15,41 @@ type NavBarProps = {
+      <StyledPagesLinks>
+        <StyledLinkToPages
+          isActive={activeLink === hrefHomePage}
+          onClick={() => handleLinkClick(hrefHomePage)}
```
do you need twice onClick for the same nav item?

Niv-Regev-Moveo committed:
Adding styles
fixed the searchBar Logic
Use UseParams inside the FavoritePage
Pokemon type Background-color is changing the color according to type of the pokemon
